File: app/src/main/java/com/shvants/runninglife/view/RunMoveView.java
package com.shvants.runninglife.view;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.ImageView;
import android.widget.TextView;

import com.shvants.runninglife.R;

import androidx.constraintlayout.widget.ConstraintLayout;


public class RunMoveView extends ConstraintLayout {

    private UserView userView;
    private ImageView avatar;
    private TextView fullName;
    private TextView beginTime;

    public RunMoveView(final Context context) {
        this(context, null);
    }

    public RunMoveView(final Context context, final AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public RunMoveView(final Context context, final AttributeSet attrs, final int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        initUserView();
    }

    private void initUserView() {
        inflate(getContext(), R.layout.move_view, this);

        avatar = findViewById(R.id.userAvatar);
        fullName = findViewById(R.id.userFullName);
        beginTime = findViewById(R.id.userLocation);
    }
}

File: app/src/main/java/com/shvants/runninglife/ui/view/RunMoveView.kt
package com.shvants.runninglife.ui.view

import android.content.Context
import android.util.AttributeSet
import android.view.View
import android.widget.TextView
import androidx.annotation.DrawableRes
import com.shvants.runninglife.R
import com.shvants.runninglife.ui.base.BaseMoveLayout
import com.shvants.runninglife.utils.Const.NULL
import com.shvants.runninglife.utils.Const.ZERO
import kotlinx.android.synthetic.main.move_view.view.*

class RunMoveView
@JvmOverloads
constructor(
        context: Context,
        attrs: AttributeSet? = NULL,
        defStyleAttr: Int = ZERO) : BaseMoveLayout(context, attrs, defStyleAttr) {

    private var tempo: TextView? = null

    override fun onViewInflated(context: Context) {
        super.userAvatar = userAvatar
        super.userFullName = userFullName
        super.startTime = startTime

        super.title = moveTitle
        super.distance = moveDistanceValue
        super.time = moveTimeValue

        tempo = moveTempoValue
    }

    fun setUserAvatar(@DrawableRes avatarId: Int): RunMoveView {
        userAvatar?.setImageResource(avatarId)

        return this
    }

    fun setUserFullName(fullName: String): RunMoveView {
        userFullName?.text = fullName

        return this
    }

    fun setStartTime(startTime: Long): RunMoveView {
        userFullName?.text = getStartTimeAsString(startTime)

        return this
    }

    private fun getStartTimeAsString(startTime: Long): CharSequence? {

    }

    override fun getLayoutResId(): Int {
        return R.layout.move_view
    }
}